pkg/addonmanager/addonclients/upgrader.go

```diff
@@ -35,6 +42,9 @@ func (f *FluxAddonClient) Upgrade(ctx context.Context, managementCluster *types.
 	if err := f.upgradeFilesAndCommit(ctx, newSpec); err != nil {
 		return nil, fmt.Errorf("failed upgrading Flux from bundles %d to bundles %d: %v", currentSpec.Bundles.Spec.Number, newSpec.Bundles.Spec.Number, err)
 	}
+	if err := f.flux.DeleteFluxSystemSecret(ctx, managementCluster, newSpec.GitOpsConfig.Spec.Flux.Github.FluxSystemNamespace); err != nil {
```

Comment:
We don't need to figure this out now, but question for the future
Will this work once flux has been upgraded to v0.23? Like imagine the user upgrades to 0.23 and we then release a new bundle with v0.24, will this same process work? Or do we need to check somehow if the secret needs to be deleted first?

Reply:
i believe so, `flux bootstrap` cmd will always generate a ecdsa-sha2-nistp384 SSH key and will update the deploy key on GitHub.